fuel-poa-coordinator/src/service.rs

```diff
@@ -56,6 +59,24 @@ use tracing::{
     warn,
 };
 
+#[async_trait::async_trait]
+pub trait BlockProducer: Send + Sync {
+    // TODO: Right now production and execution of the block is one step, but in the future,
```

Comment:
Since we are refactoring right now, would it be worth considering building out this code path right now to make it easier for future milestones. `produce_and_execute_block` is relatively simple and we could make a `produce` function which returns a block, then just call an `execute` function right away, which preserves functionality but avoids a future breaking interface change on the `BlockProducer`

Reply:
It is not refactoring mentioned in the https://github.com/FuelLabs/fuel-core/issues/809. It is simply moving from one place to another to unblock the development for me and don't create a mess in one PR=)

The comment describes the problem bad=) We can't produce the block without the execution of transactions(otherwise, it is a useless block). The comment was meant to be, "The current code produces, executes, and commits the block. But it should only produce and execute".

It is that I'm trying to fix=) This "refactoring" for [this](https://github.com/FuelLabs/fuel-core/compare/feature/block-producer-adapter...bugfix/race-condition-seal-block?expand=1) change. But without https://github.com/FuelLabs/fuel-core/pull/812 it looks ugly.